src/twisted/test/test_process.py

```diff
@@ -1199,6 +1199,115 @@ def checkTerminated(ignored):
 
         return ended
 
+    def checkSpawnProcessEnvironment(self, usePosixSpawnp, spawnKwargs, expectedEnv):
+        """
+        Shared code for testing the environment variables
+        present in the spawned process.
+
+        The spawned process serializes its environ to stderr or stdout (depending on usePTY)
+        which is checked against os.environ of the calling process.
+        """
+        p = Accumulator()
+        d = p.endedDeferred = defer.Deferred()
+
+        reactor._neverUseSpawn = not usePosixSpawnp
```

Comment:
Instead of using the global reactor,
maybe we can instantiate a separate reactor and then we will not have to care about the cleanup.

The suggestion is just an idea... the import should be at the start of the file.

```suggestion
        from twisted.internet.selectreactor import SelectReactor()
        supportingReactor = SelectReactor()
        supportingReactor._neverUseSpawn = not usePosixSpawnp
```

I think that is important not to modify the global / singleton instanced during a test.

Reply:
This isn't a safe way to do this.  Instead, there's a whole framework in twisted/internet/test/ for using per-test reactors (including many tests for `IReactorProcess`).

Reply:
Thanks for the info.

My suggestion is not ok as it only guarantees that the `spawnProcess` will work for the `SelectReactor` 

The test is not executed for all the other reactors.

I guess that these tests needs to be moved to https://github.com/twisted/twisted/blob/trunk/src/twisted/internet/test/test_process.py#L106-L113

and use `reactor = self.buildReactor()`

In this way, the tests are wrote once, but executed for any supported reactor.

Reply:
> I guess that these tests needs to be moved to https://github.com/twisted/twisted/blob/trunk/src/twisted/internet/test/test_process.py#L106-L113

Yes, exactly.

`twisted.test` is largely a trap for contributors at this point. We need to more clearly signpost the good vs. the bad ways to write tests and try to direct people to copy from the good examples (or at least, the better ones) and avoid these antipatterns.

Reply:
Thanks for suggestions. I've applied them all.